cxl institute layout

chanar opened the pull request:
- Change layout components ~too~ to cxl-*-layout

- enable lumo colors and typography

- update README.md

- add cxl-core, cxl-lumo-styles, cxl-marketing, cxl-skin

- add cxl-layout-institute component and change all layout namings to layout-*.js/scss

- add institute left icon navbar, add more components for institute (institute-content, instutute-content-header)

- add sidebar to institute layout

- add content area to institute layout

- move icon nav to cxl-institute-layout

- remove files left from massive rebase

lkraav commented:
cxl-marketing stuff needs to be kicked out of this branch.

chanar commented on packages/storybook/layouts.stories.js:
```diff
@@ -0,0 +1,290 @@
+  .add('cxl-institute-layout', () => {
+    return html`
+      <cxl-institute-layout>
+        <div id="left-nav" slot="icon-nav">
```
@chanar

lkraav replied:
I think "Outdated" label on the file means we can resolve conversation and re-open if needed in "Files changed".

chanar commented on package.json:
```diff
@@ -1,15 +1,16 @@
-  "license": "MIT",
+  "version": "0.1.0",
+  "author": "CXL",
+  "license": "UNLICENSED",
```
@chanar

chanar replied:
@lkraav didn't find what you mentioned. All your branches + master have the same package.json from upstream

lkraav replied:
Where did you look? I found it in 10 seconds

https://github.com/conversionxl/aybolit/commit/837c37935744a5834fcb6a3633bd614a1ae334fc

lkraav replied:
Still no LICENSE file?

lkraav replied:
@chanar isn't this `package.json` supposed to update as well, if you've added the correct LICENSE also to root level?

chanar replied:
possibly true

lkraav replied:
Pushed root LICENSE myself.

chanar commented on package.json:
```diff
@@ -1,15 +1,16 @@
-  "description": "Web Components library built with LitElement",
-  "author": "Serhii Kulykov <serguey.kulikov@gmail.com>",
-  "license": "MIT",
+  "version": "0.1.0",
```
@chanar

chanar commented on package.json:
```diff
@@ -1,15 +1,16 @@
 {
-  "name": "aybolit-workspace",
+  "name": "@conversionxl/cxl-core",
```
@chanar

chanar commented on package.json:
```diff
@@ -82,7 +83,6 @@
       "git add"
     ],
     "*.scss": [
-      "stylelint --syntax=scss",
```
No time to invest in list errors like "color property must be before this and that property". Need to configure it to lint only the basic things.

lkraav replied:
I will take care of the stylelint update right now.

Ordering will be a thing (alphabetic), get used to it. It's the only way the reviewers / debuggers will know to look for something that is or isn't supposed to be in a rule.

lkraav replied:
Handled in https://github.com/conversionxl/aybolit/pull/10/files#diff-ab1d354bf78d8f05312122983731ff51

chanar commented on packages/cxl-institute/scss/global.scss:
```diff
@@ -0,0 +1,211 @@
+@import "@conversionxl/cxl-ui/scss/utils/mixins";
```
I recently asked should we load everything under layout component. From this convers. I created cxl-institute and cxl-marketing that load: components, stylings, global css.

lkraav replied:
Where's the conversation link? "Something someone thought at some time" is not actionable info.

lkraav replied:
Itmw, I've learned more about global vs shadow, hence this is likely to be optimized further over time, as we see how things will play out in PHP app.

chanar commented on packages/cxl-lumo-styles/scss/colors.scss:
```diff
@@ -0,0 +1,94 @@
+  --lumo-tint-70pct: hsla(0, 0%, 100%, 0.77);
+  --lumo-tint-80pct: hsla(0, 0%, 100%, 0.84);
+  --lumo-tint-90pct: hsla(0, 0%, 100%, 0.9);
+  --lumo-tint: #fff;
```
Yes but how many times I have to say that html headings, links, etc. need to be there to apply these styles globally. And why cherry pick only that is needed? It's better to have full template so it's easier to understand what's there to use.

All those color variables are from time when Jaan applied our colors. Again why should we cherry pick. We are so much in the beginning where we most possibly need to use some of these colors but we don't know yet which ones. Let them be there to be easily adjustable and usable.

lkraav replied:
> Yes but how many times I have to say that html headings, links, etc. need to be there to apply these styles globally.

Doesn't `vaadin-lumo-styles` already do it?

lkraav replied:
> It's better to have full template so it's easier to understand what's there to use.

That's probably the job of a better integrated Lumo demo site, not staring at codebase.

Right now, "less code" trumps everything else.

lkraav replied:
Remove duplicates by tomorrow.

lkraav replied:
It doesn't look like I will get this far today, so Thursday then.

lkraav replied:
Worked it out on Slack.

chanar commented on packages/cxl-lumo-styles/scss/vaadin-button.scss:
```diff
@@ -0,0 +1,103 @@
+
+  min-width: auto !important;
+}
+:host([theme~="main-button"]) [part="label"] {
```
vaadin-button custom styles are pretty much first things done with this project. I will re-write the vaadin-button.scss properly. Don't bother to review it today.

chanar replied:
@chanar

chanar replied:
I removed currently all and left empty :host {}. No reason to delete the file as I will be updating it as required buttons are added.

chanar commented on packages/cxl-lumo-styles/scss/vaadin-details.scss:
```diff
@@ -0,0 +1,34 @@
+  [part="summary"] {
+    padding: var(--lumo-space-l) var(--lumo-space-m);
+    border-radius: 0;
+    border-top: 2px solid #e1e1e1;
```
color monster

chanar commented on packages/cxl-lumo-styles/scss/vaadin-details.scss:
```diff
@@ -0,0 +1,34 @@
+    border-top: 2px solid #e1e1e1;
+    cursor: pointer;
+
+    display: grid;
```
Need to re-try because I believe there was reason I had to go with grid.

chanar commented on packages/cxl-lumo-styles/scss/vaadin-button.scss:
```diff
@@ -0,0 +1,103 @@
+  --lumo-button-size-s: 38px !important;
+  --lumo-button-size-xs: 32px !important;
+
+  min-width: auto !important;
```
@chanar

chanar commented on packages/cxl-ui/scss/cxl-institute-layout.scss:
```diff
@@ -0,0 +1,127 @@
+  position: relative;
+  display: grid;
+  grid-template-areas: "nav sidebar content";
+  grid-template-columns: 67px 315px auto;
```
@chanar

chanar replied:
Hmm, how would you solve it with calc?

chanar commented on packages/cxl-ui/scss/cxl-institute-layout.scss:
```diff
@@ -0,0 +1,127 @@
+  grid-template-areas: "nav sidebar content";
+  grid-template-columns: 67px 315px auto;
+
+  background: #f6f6f6;
```
color monster

chanar commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,94 @@
+.vertical-line {
+  height: 1px;
+  background: #eee;
+}
```
It's like <hr> but stupider

chanar replied:
Looks like some of the html is allowed in comments.

<h1>h1</h1>
<hr>
<input type="text">

lkraav replied:
OK this was part of marketing-layout anyway, so skipping it for now.

chanar commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,94 @@
+.wrap {
+  margin: 0 auto;
+  position: relative;
+}
```
How to call a "wrapper" that is not wrap but wraps stuff?

lkraav replied:
There's a conceptual problem already by saying such things.

Something that wraps something, is always a `.wrap`.

Maybe `.wrap#some-id` to help out w/ specifics.

chanar commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,94 @@
+  position: relative;
+}
+
+@media screen and (min-width: 1088px) {
```
Somewhere commented about that these .wrap that are coming from wordpress had it's specific breaking points that didn't match with breakpoints in mixins.scss

lkraav replied:
Hmmm, no idea what it means or not, but this repo is the master MQ definition. If WP tries to say anything else, it's a theme bug to be fixed.

chanar commented on packages/cxl-ui/src/components/cxl-institute-content.js:
```diff
@@ -0,0 +1,49 @@
+
+  constructor() {
+    super();
+    this.hideSidebar = false;
```
Looks like old code. hideSidebar should go to trash. content-header component is controlling collapse now.

chanar replied:
@chanar

chanar commented on packages/cxl-ui/src/components/cxl-institute-layout.js:
```diff
@@ -0,0 +1,61 @@
+  render() {
+    return html`
+      <!-- Visible on mobile -->
+      <div id="header" part="header">
```
@chanar

chanar commented:
> cxl-marketing stuff needs to be kicked out of this branch.

@lkraav I will do it. Just a reminder that we should follow 80/20 and not try to achieve perfectionism in every way possible. We won't get anywhere like this, especially currently where MVP > perfect code.

chanar commented on .eslintrc:
```diff
@@ -9,16 +9,12 @@
     "no-restricted-globals": "off",
-    "no-underscore-dangle": "off"
+    "no-underscore-dangle": "off",
+    "camelcase": "off"
```
Just camelCase was added Tried without it and doesn't give errors. Might have been left in from testing period.

lkraav replied:
Still no answer to "why"? How come padawan hasn't needed it and we do?

chanar commented on packages/cxl-ui/scss/cxl-institute-layout.scss:
```diff
@@ -0,0 +1,127 @@
+  flex: 0 auto;
+}
+[part="header"] .logo .cxl {
+  font: italic 900 1.6rem Roboto, sans-serif;
```
You need to show how you use lumo here

lkraav commented on packages/storybook/.storybook/preview-head.html:
```diff
@@ -0,0 +1,11 @@
+<style>
+  body, html {
+    margin: 0;
```
I wonder if our layout components should normalize via https://github.com/sindresorhus/modern-normalize or somesuch.

lkraav replied:
Later, #15

lkraav commented:
> Just a reminder that we should follow 80/20 and not try to achieve perfectionism in every way possible. We won't get anywhere like this, especially currently where MVP > perfect code.

Certainly, but these first components have been so random, there's no way to build a company's future development process on it.

I've spent years and worked with tens and tens of repos by now. My QA radar is tuned to that experience, and what big repo owners a lot smarter than me have shown and required on their own stuff over years. I will most certainly not settle for a crap baseline here, for sure.

lkraav commented:
Something ungood happens to sidebar CSS when you resize Storybook viewport to mobile, then reset.

![image](https://user-images.githubusercontent.com/147228/56842056-65bf7380-6857-11e9-8396-cb3739318867.png)

lkraav commented:
@lkraav TODO

- [ ] spec WP sidebar vs widgets HTML structure
- [ ] set up icon nav WP menu ~> spec HTML structure
- [ ] mobile: header structure
- [ ] mobile: replace all "Something"-s with meaning
- [ ] mobile: icon nav vertical size - too big?
- [ ] mobile: diagnose viewport scroll vs nav elements dynamics (down: hide, up: re-appear)
- [ ] video course, lesson view
- [ ] ...

chanar commented:
Yeah, still figuring it out. Need to find a simple way to test it on`index.html`. Probably @pika/pack build does the thing.

lkraav commented:
## CXL icon nav structure prototype

* front https://conversionxli.warmpress.com/lesson/impact-of-correcting-mistakes/
* admin https://conversionxli.warmpress.com/wp-admin/nav-menus.php?menu=3265
* top level `.menu > .menu__items > .menu__item.has-children` elements are hidden, because they're containers - `.menu > .menu__items` should be styled as a `flex` / `grid` container
* `iron-icon` output actually hasn't been implemented yet, I'll see about this tomorrow
* https://conversionxl.slack.com/archives/C0G52K19U/p1556508522121500 discusses `cxl-icon-nav` custom element refactor vs shadow/light DOM CSS protection - if you agree, go for it right away in the morning cc @chanar 

```html
<nav class="menu menu--institute-icon-nav">

    <h3 class="menu__title screen-reader-text">
			cxl-2019/institute-icon-nav		</h3>

    <ul class="menu__items">
        <li id="menu-item-1031803" class="menu__item has-children "><a href="#" class="menu__link">TOP</a>
            <ul class="menu__sub-menu">
                <li id="menu-item-1031802" class="menu__item "><a href="https://conversionxli.warmpress.com/practitioners/" class="menu__link"><strong>C<em>XL</em></strong></a></li>
                <li id="menu-item-1031801" class="menu__item "><a class="menu__link">cxl:nav-trainings</a></li>
                <li id="menu-item-1031804" class="menu__item "><a class="menu__link">cxl:nav-roadmap</a></li>
                <li id="menu-item-1031805" class="menu__item "><a href="#" class="menu__link">cxl:nav-discover</a></li>
            </ul>
        </li>
        <li id="menu-item-1031806" class="menu__item has-children "><a class="menu__link">BOTTOM</a>
            <ul class="menu__sub-menu">
                <li id="menu-item-1031808" class="menu__item "><a href="https://conversionxli.warmpress.com/my-account/" class="menu__link">lumo:user</a></li>
            </ul>
        </li>
    </ul>
</nav>
```

lkraav commented:
# Stylelint CXL rules

See https://github.com/conversionxl/aybolit/tree/wip-leho-cxl-institute-layout for kickoff

* Alphabetic order (easy to remember), we'll deal with more complex approaches later
* Single lines (easier to grasp structural problems, but doesn't work yet, commit hooks overwrite something)

@chanar `<cxl-institute-layout>` is the first component to introduce custom CSS, so you could squash my wip branch `.stylelintrc` copy into your main commit, and run `stylelint --fix` it will auto-arrange everything, then amed / re-squash that result as well.

chanar commented:
> https://conversionxl.slack.com/archives/C0G52K19U/p1556508522121500 discusses cxl-icon-nav custom element refactor vs shadow/light DOM CSS protection - if you agree, go for it right away in the morning cc @chanar

It wouldn't be much different from just using id or class name on the <nav> element. Global css is needed in both ways.

chanar commented:
icon nav needs two different icons and name of menu item for mobile

```
<iron-icon class="size-l" icon="cxl:nav-trainings"></iron-icon>
<iron-icon class="size-l" icon="cxl:nav-trainings-active"></iron-icon>
<span>Trainings</span>
```

chanar commented:
@lkraav not much I can do more https://github.com/conversionxl/aybolit/pull/10/commits/e884d8243b8513abbbf176e08df44f43fc1be55f

Waiting for menu markup to be finished. Can we use logo separately? We have component for logo cxl-logo

lkraav commented:
I had a couple of distractions today and didn't quite get back on track here.

Couple of notes for tomorrow:

* would be nice to have a <cxl-institute-layout> prototype video implementation back

* <cxl-institute-layout> secondary (serif) font seems to be missing, storybook app should be responsible for font provision (as php theme will be)

* avoid stylelinting web-padawan/aybolit upstream packages, that's future merge conflict hell waiting to happen - instead we will probably add additional `.stylelintrc` files in `packages/{core,white-label}/` directories, so they will always maintain their upstream rules - restore originals, re-force-push

* maybe start tuning up <cxl-marketing-layout> markup based on recent learnings, etc? I look to get back on track here tomorrow.

chanar commented:
If we want to get rid of global embed video player styles, I would do `<cxl-institute-content-video src="...">` https://github.com/conversionxl/aybolit/pull/10/commits/0501ddc430d75b3667c15a2230cfdb068effa04b

chanar commented on packages/cxl-institute/scss/global.scss:
```diff
@@ -0,0 +1,298 @@
+  }
+}
+
+@include respond-below(small) {
```
Yes, will move mq-s inside their actual elements. About the mobile-first approach, isn't it easier to manage more complicated (desktop) css in top and then add properties to make a change towards mobile?

chanar commented on packages/cxl-ui/scss/cxl-institute-content.scss:
```diff
@@ -0,0 +1,68 @@
+
+  &[media-size="large"] {
+    margin: 0 auto;
+    max-width: 1024px;
```
Currently changed in the same file. Not sure yet how would I configure .wrap for it. Need to test current wrap in layout

chanar commented on packages/cxl-ui/src/components/cxl-institute-content.js:
```diff
@@ -0,0 +1,40 @@
+    return html`
+      <slot name="content-header"></slot>
+      <slot name="content-video"></slot>
+      <div class="wrapper">
```
Currently stays as I won't have time to figure it out. But would do it `with media-size="large"` comment

lkraav replied:
I will investigate `.wrap` strategy in PHP theme.

chanar commented:
> * No magic numbers.
> * No "lint-only" modifying upstream SCSS files.
> * Make sure your own things are linted! Restore `package.json` for `stylelint`.
> * MQ-s moved inside elements.


- Removed magic numbers. Added stylelint plugin `stylelint-declaration-use-variable`. It's not perfect because I cannot ignore 0 and auto values. Therefore I'm keeping padding- and margin- ignored by default. When needed to run for large amount of scss these can be moved out of the ignore object `"sh-waqar/declaration-use-variable": [["/color/", "font-size", "line-height", { ignoreValues: ["/padding-/", "/margin-/"] }]]`
- reverted core and white-label to master and added .stylelintignore file to keep these paths out of the linting.
- moved mqs inside classes

I have to go now

lkraav commented on .stylelintignore:
```diff
@@ -0,0 +1,3 @@
+packages/core/scss
+packages/white-label/scss
+packages/cxl-lumo-styles/scss/typography.scss
```
Smart :+1: yeah we can exclude upstream for time being, I don't see us doing much upstream commits in this branch.

lkraav commented on packages/cxl-institute/scss/global.scss:
```diff
@@ -0,0 +1,299 @@
+  }
+}
+
+.icon-nav {
```
I don't get it, old styles are still sticking around? Is this the same as `.menu--institute-icon-nav`?

chanar replied:
Yes, I have kept the old one with stories extra layout until icon nav with WP markup gets completed.

chanar replied:
@chanar

lkraav commented on packages/cxl-lumo-styles/package.json:
```diff
@@ -1,15 +1,34 @@
-  "author": "CXL <leho@conversionxl.com>",
-  "license": "SEE LICENSE IN LICENSE.md",
+  "author": "CXL",
+  "license": "UNLICENSED",
```
Careless license revert?

chanar replied:
@chanar

chanar commented on package.json:
```diff
@@ -76,15 +77,22 @@
-      "stylelint --syntax=scss",
-      "prettier --single-quote=false --write",
-      "git add"
+    "linters": {
```
It's still lint staged but https://github.com/okonet/lint-staged#packagejson-example-with-ignore-option

lkraav replied:
OK I see it now :+1:

lkraav commented on packages/cxl-lumo-styles/scss/vaadin-tabs.scss:
```diff
@@ -0,0 +1,20 @@
+@import "@conversionxl/cxl-ui/scss/utils/mixins";
+
+:host([theme~="institute-layout-tabs"]) [part="tabs"] {
```
Prefix all `theme` attributes `cxl`.

I should be able to grep through codebase to find all our custom `theme` attributes.

lkraav commented on packages/cxl-lumo-styles/src/templates/colors.js:
```diff
@@ -0,0 +1,11 @@
+const $template = document.createElement('template');
+
+$template.innerHTML = `
+<custom-style>
```
So this ended up needing `<style include="lumo-color">` etc.

lkraav commented on packages/cxl-lumo-styles/src/templates/typography.js:
```diff
@@ -0,0 +1,11 @@
+
+$template.innerHTML = `
+<custom-style>
+  <style>${lumoTypography}</style>
```
`<style include="lumo-typography">`

lkraav commented on packages/cxl-lumo-styles/scss/vaadin-button.scss:
```diff
@@ -0,0 +1 @@
+@import "@conversionxl/cxl-ui/scss/utils/mixins";
```
Buttons should never import things like media queries.

Slack call - make container navbar respond, override `:host` w/ something like `vaadin-buttom[theme=]`.

lkraav commented on packages/cxl-ui/scss/cxl-institute-content-header.scss:
```diff
@@ -0,0 +1,63 @@
+
+:host {
+  border-bottom: 1px solid rgb(238, 238, 238);
+  display: grid;
```
It's one-dimensional. `flex`

lkraav commented on packages/cxl-ui/scss/cxl-institute-content.scss:
```diff
@@ -0,0 +1,64 @@
+@import "./utils/mixins";
```
Follow https://github.com/justintadlock/mythic/tree/master/resources/scss structure.

chanar replied:
He does have defined break points but using direct `min-width `etc. although we have been using `respond-above(small)` and it also has `respond-between(small, medium)`. I'm currently not seeing the value of his scss structure + https://github.com/justintadlock/mythic/blob/master/resources/scss/settings/_breaks.scss

On another subject, if we would like to convert our styles to mobile-first, what would be the best steps?

lkraav replied:
We can just adopt the structure, directory / file naming for one.

Going with `sass-mq` should handle the rest of actual meat functionality for us.

Min-width-only is better to start with, it keeps out hacky bs. When a real use case arises to the type of "I really can't do this without some `between` type thing", I'd like to see. It may be a conceptual problem that needs to be fixed instead.

Mobile-first requires a bit of a refactor (re-ordering of cascade, and re-assigning mq-s), but we should do it before distributing this framework to any new people.

chanar replied:
Yes, lets do mobile first refactor asap. Question is, before or after institute merge to master?

chanar replied:
Nah, I will do it right now

lkraav commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,85 @@
+  padding-left: calc(var(--lumo-space-m) / 2);
+  padding-right: calc(var(--lumo-space-m) / 2);
+
+  @media screen and (min-width: 1088px) {
```
Replace w/ master media queries defined here?

chanar replied:
Don't know what is the decision here. These are WP breakpoints (should use the current mq mixins though) but the breakpoints are different

Our project breakpoints:
```
small: 724px,
medium: 992px,
large: 1279px,
largehd: 1600px
```

and compared to WP .wrap breakpoints that we changed to (before it was .container)
<del>```small: 724px```</del>
```
medium: 1088
large: 1280px,
largehd: 1472px
```

lkraav replied:
What is "WP .wrap" even? Screenshot?

I think current WP whatever doesn't matter, it will be adjusted to what we do here.

lkraav commented:
Go for it right away, if you can. I can work through it in the weekend then.


On 3 May 2019 11:52:39 Janar Palk <notifications@github.com> wrote:

> chanar commented on this pull request.
>
>
>
>> @@ -0,0 +1,64 @@
> +@import "./utils/mixins";
>
> Yes, lets do mobile first refactor asap. Question is, before or after 
> institute merge to master?
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/conversionxl/aybolit/pull/10#discussion_r280732989

lkraav commented on packages/cxl-institute/scss/global.scss:
```diff
@@ -96,20 +96,20 @@ body {
-    @include mq($until: wide) {
-      height: 400px;
+    @include mq($from: wide) {
+      height: 500px;
```
What sense does "going in reverse" make?

"Mobile first" should mean "smaller to bigger" across the specturm.

lkraav commented on packages/cxl-ui/scss/cxl-institute-content.scss:
```diff
@@ -51,14 +49,14 @@
-  padding: var(--lumo-space-xl) var(--lumo-space-m);
+  padding: var(--lumo-space-m) var(--lumo-space-m);
 
   &[media-size="large"] {
```
Should these match mq breakpoint names?

lkraav commented on packages/cxl-institute/package.json:
```diff
@@ -0,0 +1,42 @@
+    "@vaadin/vaadin-button": "^2.1.5",
+    "@vaadin/vaadin-details": "^1.0.1",
+    "lit-element": "^2.1.0",
+    "sass-mq": "^5.0.0"
```
@chanar is `sass-mq` actually a devDependency, or? How is it involved in post-compile runtime?

chanar replied:
Yes correct, this should be devDep. Same thing with cxl-ui package.

lkraav commented on packages/cxl-ui/src/components/cxl-institute-layout.js:
```diff
@@ -0,0 +1,66 @@
+        <slot name="icon-nav"></slot>
+      </div>
+
+      <div id="sidebar" part="sidebar">
```
@chanar sidebar was always designated to by `<aside>` element.

Until an exact custom element strategy is decided, make the change.

chanar replied:
Are you sure? Any reference?

"The <aside> element is used to identify content that is related to the primary content of the webpage, but does not constitute the primary content of the page. Author information, related links, related content, and advertisements are exampes of content that may be found in an aside element."

https://aastudio.fr/Sidebar-and-Aside-are-different.html

I have never seen aside being used in context of eg. <sidebar>sidebar here</sidebar>

lkraav replied:
Excellent article, and it supports my request.

> The aside tag outside a `<section>` or `<article>` tag
> If <aside> is used outside a <section> or an <article> tags - directly within <body> for instance -, it carries a content related to the site such as : tweets from its author, additional navigation, last blog’s posts, last comments, …

Institute app sidebar only carries information related to content area.

Our main nav is already sitting outside of sidebar. Inside, this would indeed violate `<aside>` content principles.

lkraav commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,61 @@
+
+  @include mq($from: desktop) {
+    max-width: 960px;
+    width: 960px;
```
No hard-coded `width` values, this makes `.wrap` only usable in specific outer container cases, and that's not the intention.

lkraav commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,61 @@
+
+  @include mq($from: wide) {
+    max-width: 1152px;
+    width: 1152px;
```
Same as above.

lkraav commented on packages/cxl-ui/scss/cxl-theme.scss:
```diff
@@ -0,0 +1,61 @@
+
+  @include mq($from: desktop) {
+    max-width: $wrap_desktop;
+    width: $wrap_desktop;
```
Didn't we just remove hardcoded `width` values?

lkraav commented on packages/cxl-ui/scss/utils/_mixins.scss:
```diff
@@ -0,0 +1,49 @@
+  largehd: 1600px
+);
+
+@mixin respond-above($breakpoint) {
```
With `sass-mq`, can these mixins now be trashed?

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav.js:
```diff
@@ -0,0 +1,24 @@
+
+  render() {
+    return html`
+      <ul>
```
`ul`? https://www.w3.org/TR/html50/grouping-content.html#the-ul-element says only `li`, `script`, or `template` elements are allowed in `ul`.

lkraav replied:
If need a container, wouldn't it be a `<nav>`?

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav-item.js:
```diff
@@ -0,0 +1,33 @@
+
+  render() {
+    return html`
+      <li>
```
`ul > li` spec doesn't jump across custom element element barriers - `li` is only legit as a direct child.

chanar replied:
Hmm didn't think of that. Going to try it with vaadin-list-box, same thing with `<cxl-institute-sidebar-item>`

chanar replied:
I have used to do it in Vue where shadow dom doesn't exist.

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav-item.js:
```diff
@@ -0,0 +1,33 @@
+class CXLIconNavItemElement extends LitElement {
+  static get properties() {
+    return {
+      href: {
```
Properties is a good idea, it could also take the icon identifier, I think.

lkraav commented on packages/cxl-ui/src/components/cxl-institute-layout.js:
```diff
@@ -0,0 +1,66 @@
+        <slot name="sidebar-content"></slot>
+      </div>
+
+      <div id="content" part="content" page-visible>
```
@chanar OK with you to drop `id` attributes, and continue w/ `part`?

Seems to be Vaadin model, right? Since why would you do ID-s inside Shadow DOM.

lkraav replied:
Got confirmation from Vaadin https://gitter.im/vaadin/web-components?at=5cdbf2795a1d435d4645d524 so dropping ID-s where not specifically needed.

lkraav commented on packages/cxl-ui/src/components/cxl-institute-layout.js:
```diff
@@ -0,0 +1,66 @@
+
+      <div id="content" part="content" page-visible>
+        <slot></slot>
+      </div>
```
More content things, I think this should be a `<main>` element https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main

lkraav commented on packages/cxl-ui/scss/utils/_mq.scss:
```diff
@@ -0,0 +1,34 @@
+  tablet: 740px,
+  desktop: 980px,
+  wide: 1300px,
+  /* Tweakpoints*/ desktopad: 810px,
```
What is Tweakpoints? I'm inclined to remove it, no usage anywhere.

lkraav commented on packages/cxl-ui/scss/utils/_mq.scss:
```diff
@@ -0,0 +1,34 @@
+
+// Wrap width
+$wrap_desktop: 960px;
+$wrap_wide: 1152px;
```
Wouldn't this be some kind of formula?

`calc( $mq-breakpoints['mobile'] - ( var(--lumo-sizing-xl * 2 ) )` or similar?

lkraav replied:
I also don't understand why we need mq-specific variable values at all.

lkraav commented on packages/cxl-institute/scss/global.scss:
```diff
@@ -0,0 +1,75 @@
+}
+
+// TODO: Any better solution?
+.sidebar-content {
```
Let's discuss TODO status here?

lkraav commented on packages/cxl-institute/scss/global.scss:
```diff
@@ -0,0 +1,75 @@
+  height: 100%;
+  width: 100%;
+
+  .extra-heading {
```
TODO remove when sidebar markup spec completed

lkraav commented on packages/cxl-ui/scss/cxl-institute-content.scss:
```diff
@@ -0,0 +1,65 @@
+
+  &[media-size="wide"] {
+    margin: 0 auto;
+    max-width: 35em;
```
This should probably be derived from a breakpoint value `calc()`.

lkraav replied:
I was wrong, but I will move this into a variable at the top, so it's not a totally obscure thing.

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav-item.js:
```diff
@@ -0,0 +1,32 @@
+  render() {
+    return html`
+      <a href="${this.icon}">
+        <iron-icon class="icon size-l" icon="${this.icon}"></iron-icon>
```
Need `active` state implementation.

lkraav replied:
WP will output a set of relevant classes on `<vaadin-tab>`, it's too much work to re-architect that, so we should try to make it work.

Discuss when we come to it.

lkraav commented on packages/cxl-ui/src/components/cxl-institute-layout.js:
```diff
@@ -0,0 +1,67 @@
+        <vaadin-tab @click="${() => this.changeTab('content')}">Something here</vaadin-tab>
+      </vaadin-tabs>
+
+      <div part="icon-nav">
```
This fairly certainly has to be `<nav>`, pushing this change now.

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav-item.js:
```diff
@@ -0,0 +1,32 @@
+
+  render() {
+    return html`
+      <a href="${this.icon}">
```
@chanar ouch, this has to be `${this.href}`

I'd actually prefer this to be fetched from element content. `<cxl-icon-nav-item href="#">cxl:training</cxl-icon-nav-item>`. Are there any drawbacks?

lkraav replied:
`${this.innerHTML}` seems to work without side-effects.

lkraav replied:
`href` attribute needs to be conditional. Not always guaranteed to be there (admin menu editor allows empty), and we should not render `undefined` values.

`<a ?href="${this.innerHTML}">`

lkraav replied:
https://github.com/Polymer/lit-html/issues/625 actually points out `ifDefined()` is the correct solution.

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav.js:
```diff
@@ -0,0 +1,46 @@
+    } else {
+      tabsElement.setAttribute('orientation', 'vertical');
+    }
+  }
```
Avoid if/else dances.

```js
let orientation = 'vertical';

if ( width < 720 ) {
  orientation = 'horizontal';
}

tabsElement.setAttribute( 'orientation', orientation );
```

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav-item.js:
```diff
@@ -0,0 +1,32 @@
+  render() {
+    return html`
+      <a href="${this.icon}">
+        <iron-icon class="icon size-l" icon="${this.icon}"></iron-icon>
```
I'd actually prefer icons to be fetched from element content. `<cxl-icon-nav-item href="#">cxl:training</cxl-icon-nav-item>`. Are there any drawbacks?

`${this.innerHTML}` seems to work without side-effects.

lkraav replied:
@chanar I force-pushed a set of updates.

chanar replied:
Hmm, lets then use element content for initials (text)/image (imgsrc)/icon (icon name). and just use content-type="imagesrc/text/icon"

lkraav replied:
This proposal is hard to decipher. Can you provide specific markup examples for every member of our scenario set?

chanar replied:
```
  render() {
    return html`
      <a href=${ifDefined(this.href)}>
        ${this.getNavItemContent}
      </a>
    `;
  }

  getNavItemContent() {

    switch(this.contentType) {
      case 'image':
        return html`<img src="${this.innerHTML}" />`
        break;
      case 'text':
        return html`<span>${$this.innerHTML}</span>`
        break;
      default:
        return html`<iron-icon class="icon size-l" icon="${this.innerHTML}"></iron-icon>`
    }

  }
```

chanar replied:
```
<vaadin-tab>
            <cxl-icon-nav-item href="#" content-type="icon">cxl:nav-discover</cxl-icon-nav-item>
          </vaadin-tab>
```

lkraav replied:
Nice in theory, but the problem with this is requiring PHP implementation of content-type.

Could you make this work via `<vaadin-tab class>` container attribute? It is difficult to adjust anything deeper than that quickly.

chanar replied:
So instead content-type="icon" I would make 

`<vaadin-tab class="icon/image/text">`

How are you planning to add href="" to cxl-icon-nav-item? I'm actually not sure how easy it would be to remove preventDefault() from vaadin-tab (so the inner link gets clicked)

lkraav replied:
I configured staging menu CXL logo menu item to output `<vaadin-tab id="menu-item-1031802" class="menu__item menu__item--type-html">`

`menu__item--type-html` per https://themehybrid.com/weblog/why-mythic-uses-bem

lkraav commented on packages/storybook/layouts.stories.js:
```diff
@@ -0,0 +1,546 @@
+            <vaadin-accordion-panel theme="cxl-sidebar-panel">
+              <div slot="summary">
+                <vaadin-item theme="cxl-sidebar-panel-summary">
+                  <label class="label">Track #1</label>Google Analytics
```
@chanar `<label class="label">`... uh why class this?

lkraav replied:
"Google Analytics" title is missing a header. `<h4 class="entry__title">` to start.

lkraav commented on packages/cxl-ui/src/components/cxl-icon-nav-item.js:
```diff
@@ -0,0 +1,44 @@
+    return html`
+      <a href=${ifDefined(this.href)}>
+        ${this.navItemContent}
+      </a>
```
We will need an optional clarity `<label>` strategy for icons, because people are really bad at remembering what icons mean.

We currently render `title` attribute already, can use it: `<cxl-icon-nav-item title="Active Training">cxl:nav-trainings</cxl-icon-nav-item>`

```html
<a href>
  ${this.navItemContent}
  <label>${this.title}</label>
</a>
```

lkraav commented on packages/cxl-institute/src/index.js:
```diff
@@ -0,0 +1,23 @@
+import '@conversionxl/cxl-ui/src/components/cxl-institute-sidebar-items.js';
+import '@conversionxl/cxl-ui/src/components/cxl-institute-sidebar-item.js';
+import '@conversionxl/cxl-ui/src/components/cxl-icon-nav.js';
+import '@conversionxl/cxl-ui/src/components/cxl-icon-nav-item.js';
```
`cxl-icon-nav` should take care of importing item.

lkraav commented on packages/storybook/templates/cxl-vaadin-accordion.js:
```diff
@@ -0,0 +1,78 @@
+
+export default html`
+  <cxl-vaadin-accordion>
+    <vaadin-accordion-panel theme="cxl-sidebar-panel">
```
@chanar there is no matching theme style module. Is this attribute an error?

chanar replied:
`<vaadin-accordion-panel>` uses vaadin-details therefore styling must be done through vaadin-details template

chanar replied:
Also, it adds theme set for vaadin-accordion-panel to vaadin-details

lkraav replied:
Looks like this explains theme attribute propagation a bit https://vaadin.com/forum/thread/17596415/

But in the markup below, there is no `<vaadin-details>` element, it's been refactored out.

Another vote for refactoring the current untrackable styling approach. I'm working on it right now.

lkraav replied:
Mkay, new learnings: `class AccordionPanelElement extends Vaadin.DetailsElement`

lkraav replied:
In refactor, I went with `theme-for="vaadin-accordion-panel"` beacuse that's the actual element, and it seems to work fine for now.

lkraav commented on packages/cxl-lumo-styles/src/templates/vaadin-details.js:
```diff
@@ -0,0 +1,12 @@
+  <template>
+    <style>${lumoVaadinDetails}</style>
+  </template>
+</custom-style>`;
```
Yo, wrong closing tag? Shouldn't this be `</dom-module>`?

Same for almost everywhere else.

But I'm about done refactoring this, so maybe no immediate additional action needed.

lkraav commented on packages/cxl-institute/src/index.js:
```diff
@@ -0,0 +1,23 @@
+
+// Custom components.
+import '@conversionxl/cxl-ui/src/components/cxl-institute-layout';
+import '@conversionxl/cxl-ui/src/components/cxl-minidegree-progress-widget/cxl-vaadin-accordion';
```
Why is this import needed separately here?

lkraav commented on packages/storybook/templates/cxl-minidegree-progress-widget.js:
```diff
@@ -0,0 +1,82 @@
+import { html } from 'lit-html';
+
+export default html`
+<cxl-minidegree-progress-widget id="post-12067">
```
We should have all templates indented the same, see `templates/cxl-institute-content-header.js` right above.

It will create a bigger diff in the commit, but that doesn't matter right now, as a bunch of things will get squashed anyway.
